services/api/src/resources/project/resolvers.ts

```diff
@@ -629,8 +629,54 @@ export const updateProject: ResolverFn = async (
 
   const oldProject = await Helpers(sqlClientPool).getProjectById(id);
 
-  // TODO If the privateKey changes, automatically remove the old one from the
+  // If the privateKey is changed, automatically remove the old one from the
   // default user and link the new one.
+  if (patch.privateKey) {
+    let keyPair: any = {};
+    try {
+      const privateKey = sshpk.parsePrivateKey(R.prop('privateKey', patch))
+      const publicKey = privateKey.toPublic();
+
+      keyPair = {
+        ...keyPair,
+        private: R.replace(/\n/g, '\n', privateKey.toString('openssh')),
+        public: publicKey.toString()
+      };
+
+      const keyParts = keyPair.public.split(' ');
+
+      try {
+        // since public keys can only be associated to a single user
+        // if a polysite tries for some reason uses the same key across all the polysites
+        // then only 1 of the default-users will be able to hold the public key
+        // ideally each project should have its own key and polysite repos should
+        // have the public key for each project added to it for pulls
```

Comment:
yeah im not sure. i wrote it because it was the motivation for this change. im happy for it to just live on in this PRs description too if you dont want it in the code base